metrics/comet/comet.py

```diff
@@ -0,0 +1,138 @@
+# limitations under the License.
+""" COMET metric.
+
+Requirements:
+pip install unbabel-comet
+
+Usage:
+
+```python
+from datasets import load_metric
+comet_metric = load_metric('metrics/comet/comet.py')
+#comet_metric = load_metric('comet')
+#comet_metric = load_metric('comet', 'wmt-large-hter-estimator')
+
+
+source = ["Dem Feuer konnte Einhalt geboten werden", "Schulen und Kindergärten wurden eröffnet."]
+hypothesis = ["The fire could be stopped", "Schools and kindergartens were open"]
+reference = ["They were able to control the fire.", "Schools and kindergartens opened"]
+
+predictions = comet_metric.compute(predictions=hypothesis, references=reference, sources=source)
+predictions['scores']
+```
+"""
+
+import os
+from logging import getLogger
+
+from comet.models import download_model, load_checkpoint
+
+import datasets
+
+
+logger = getLogger(__name__)
+
+_CITATION1 = """\
+@inproceedings{rei-etal-2020-comet,
+   title = "{COMET}: A Neural Framework for {MT} Evaluation",
+   author = "Rei, Ricardo  and
+      Stewart, Craig  and
+      Farinha, Ana C  and
+      Lavie, Alon",
+   booktitle = "Proceedings of the 2020 Conference on Empirical Methods in Natural Language Processing (EMNLP)",
+   month = nov,
+   year = "2020",
+   address = "Online",
+   publisher = "Association for Computational Linguistics",
+   url = "https://www.aclweb.org/anthology/2020.emnlp-main.213",
+   pages = "2685--2702",
+}
+"""
+
+_CITATION2 = """\
+@inproceedings{rei-EtAl:2020:WMT,
+   author    = {Rei, Ricardo  and  Stewart, Craig  and  Farinha, Ana C  and  Lavie, Alon},
+   title     = {Unbabel's Participation in the WMT20 Metrics Shared Task},
+   booktitle      = {Proceedings of the Fifth Conference on Machine Translation},
+   month          = {November},
+   year           = {2020},
+   address        = {Online},
+   publisher      = {Association for Computational Linguistics},
+   pages     = {909--918},
+}
+"""
+
+
+_DESCRIPTION = """\
+Crosslingual Optimized Metric for Evaluation of Translation (COMET) is an open-source framework used to train Machine Translation metrics that achieve high levels of correlation with different types of human judgments (HTER, DA's or MQM).
+With the release of the framework the authors also released fully trained models that were used to compete in the WMT20 Metrics Shared Task achieving SOTA in that years competition.
+
+See the [README.md] file at https://unbabel.github.io/COMET/html/models.html for more information.
+"""
+
+_KWARGS_DESCRIPTION = """
+BLEURT score.
+
+Args:
+
+`sources` (list of str): Source sentences
+`predictions` (list of str): candidate translations
+`references` (list of str): reference translations
+`cuda` (bool): If set to True, runs COMET using GPU
+`show_progress` (bool): Shows progress
+
+
+`model`: COMET model to be used. Will default to `wmt-large-da-estimator-1719` if None.
+
+Returns:
+    `predictions`: List of dictionaries with `src`, `mt`, `ref` and `score`.
```

Comment:
Following the output of the `_compute` method I believe `predictions` should be renamed to `samples`:

```suggestion
    `samples`: List of dictionaries with `src`, `mt`, `ref` and `score`.
```